spring-pulsar-docs/src/main/asciidoc/pulsar.adoc

```diff
@@ -336,6 +340,44 @@ public ConcurrentPulsarMessageListenerContainer(PulsarConsumerFactory<? super T>
 Concurrency of more than `1` is only allowed on non-exclusive subscriptions (`failover`, `shared` and `key-shared`).
 You can only have the default `1` for concurrency when you have an exclusive subscription mode.
 
+Here is an example of enabling `concurrency` through the `PulsarListener` annotation for a `failover` subscription.
+
+====
+[source, java]
+----
+@PulsarListener(topics = "my-topic", subscriptionName = "subscription-1",
+				subscriptionType = SubscriptionType.Failover, concurrency = "3")
+void listen(String message, Consumer<String> consumer) {
+    ...
+    System.out.println("Current Thread: " + Thread.currentThread().getName());
+    System.out.println("Current Consumer: " + consumer.getConsumerName());
+}
+----
+====
+
+In the above listener, it is assumed that the topic `my-topic` has 3 partitions.
+If it is a non-partitioned topic, then having concurrency set to `3`, will not do anything, you will simply get two idle consumers in addition to the main active one.
+If the topic is over partitioned, then messages will be load balanced across the consumers that the container creates.
+If you run this `PulsarListener`, you will see that messages from different partitions will be consumed through different consumers as implied by the thread name and consumer names printouts in the example code above.
+When using the `Failover` subscription this way on partitioned topics, Pulsar guarantees message ordering.
+
+Here is another example of `PulsarListener`, but with `Shared` subscription and `concurrency` enabled.
+
+====
+[source, java]
+----
+@PulsarListener(topics = "my-topic", subscriptionName = "subscription-1",
+				subscriptionType = SubscriptionType.Shared, concurrency = "5")
+void listen(String message) {
+    ...
+}
+----
+====
+
+In the example above, the `PulsarListener` creates 5 different consumers (once again, we are assuming that the topic has 5 partitions).
+Keep in mind that, in this version, there is no message ordering as `Shared` subscriptions do not guarantee any message ordering in Pulsar.
```

Comment:
`here is no message ordering as `Shared` subscriptions do not guarantee any message ordering in Pulsar.`

This may be worth bolding or saying loudly here. It is a big fact and great thing to drill into their minds. Same as the previous non-sahred section where it IS ordered. 

Wdyt?